src/test/article.spec.js

```diff
@@ -0,0 +1,59 @@
+import chai, { expect } from 'chai';
+import chaiHttp from 'chai-http';
+import sinon from 'sinon';
+import app from '../index';
+import BaseRepository from '../repository/base.repository';
+import { createUser, createArticle, generateArticle } from './utils/helpers';
+import db from '../database/models';
+import helper from '../helpers/utils';
+
+const ARTICLES_API = '/api/v1/articles';
+
+chai.use(chaiHttp);
+
+const server = () => chai.request(app);
+
+describe('GET api/v1/articles', () => {
+  beforeEach(async () => {
+    await db.Article.destroy({ cascade: true, truncate: true });
+    await db.User.destroy({ cascade: true, truncate: true });
+  });
+
+  it('should get a list of all articles', async () => {
+    const firstUser = await createUser();
+    const secondUser = await createUser();
+    const theArticle = await createArticle(
+      await generateArticle({ authorId: secondUser.id })
+    );
+
+    const numberOfArticles = await BaseRepository.findAndCountAll(db.Article);
+    const token = helper.jwtSigner(firstUser);
+    expect(numberOfArticles.count).to.equal(1);
+
+    const res = await server()
+      .get(`${ARTICLES_API}`)
+      .set('token', token);
+    expect(res.status).to.equal(200);
+    expect(res.body.data).to.be.an('array');
+    expect(res.body.data).to.have.lengthOf(1);
+    expect(res.body.data[0].id).to.equal(theArticle.id);
+    expect(res.body.data[0].authorId).to.equal(secondUser.id);
+    expect(res.body.data[0].title).to.equal(theArticle.title);
+    expect(res.body.data[0].body).to.equal(theArticle.body);
+    expect(res.body.data[0].image).to.equal(theArticle.image);
+  });
+
+  it('should return error if database error occurs', done => {
+    const findAllStub = sinon.stub(BaseRepository, 'findAndCountAll');
+    findAllStub.rejects();
+    const userUrl = '/api/v1/articles';
+    chai
+      .request(app)
+      .get(userUrl)
+      .end((err, res) => {
+        expect(res.status).to.equal(500);
+        findAllStub.restore();
+        done();
+      });
+  });
+});
```

Comment:
The test should include tests for pagination also not only the get requests for articles